### PR TITLE
Update doc and fixed reference links

### PR DESCRIPTION
**Pull request by karan6181**
## Description
- Update doc and fixed reference links

### CODE REVIEW

File: docs/source/getting_started/quick_start.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # 🚀 Quick Start
 
-Start training your model with the Streaming dataset in a couple of steps!
+Start training your model with the Streaming dataset in a few steps!
 
-- Convert your raw dataset into a supported streaming format, for example, `mds` format.
+1. Convert your raw dataset into one of our supported streaming format, for example, `mds` (Mosaic Data Shard) format.
```
Comment (knighton) [inline docs/source/getting_started/quick_start.md:5]: formats

File: docs/source/getting_started/user_guide.md [hunk 2]
```diff
@@ -1,14 +1,25 @@
 # 🖼️ User Guide
 
-At a high level, one needs to create a dataset files compatible with a streaming and then load the same dataset files using {class}`streaming.Dataset` class.
+At a very high level, one needs to convert a raw datasets into a streaming format files and then use the same streaming format files using {class}`streaming.Dataset` class for model training.
```
Comment (knighton) [inline docs/source/getting_started/user_guide.md:3]: convert a raw dataset into streaming format files

File: docs/source/getting_started/user_guide.md [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
 
-Streaming supports different dataset writer based on your need such as {class}`streaming.MDSWriter`, {class}`streaming.CSVWriter`, {class}`streaming.JSONWriter`, {class}`streaming.TSVWriter`, and {class}`streaming.XSVWriter`. The {class}`streaming.MDSWriter` write the dataset into `.mds` extension, the {class}`streaming.CSVWriter` write the content in `.csv` format and so on. For the more information about writer and its parameters, look at the API reference doc.
+Streaming supports different dataset writer based on your need to for conversion of raw datasets into a streaming format such as
```
Comment (knighton) [inline docs/source/getting_started/user_guide.md:5]: writers

File: docs/source/getting_started/user_guide.md [hunk 4]
```diff
@@ -1,14 +1,25 @@
...
+- {class}`streaming.MDSWriter`: Writes the dataset into `.mds` (Mosaic Data Shard) extension. It supports various encoding/decoding formats(`str`, `int`, `bytes`, `jpeg`, `png`, `pil`, `pkl`, and `json`) which converts the data from that format to bytes and vice-versa.
+- {class}`streaming.CSVWriter`: Writes the dataset into `.csv` extension. It supports various encoding/decoding formats(`str`, `int`, and `float`) which converts the data from that format to string and vice-versa.
+- {class}`streaming.JSONWriter`: Writes the dataset into `.json` extension. It supports various encoding/decoding formats(`str`, `int`, and `float`).
+- {class}`streaming.TSVWriter`: Writes the dataset into `.tsv` extension. It supports various encoding/decoding formats(`str`, `int`, and `float`) which converts the data from that format to string and vice-versa.
+- {class}`streaming.XSVWriter`: Writes the dataset into `.xsv` extension. It supports various encoding/decoding formats(`str`, `int`, and `float`) which converts the data from that format to string and vice-versa.
```
Comment (knighton) [inline docs/source/getting_started/user_guide.md:10]: probably spell out what these ("xsv", "csv", "tsv") mean (comma-separated, tab-separated, etc)

File: docs/source/getting_started/user_guide.md [hunk 5]
```diff
@@ -1,14 +1,25 @@
...
+
+Streaming supports various dataset compression formats(Brotli, Bzip2, Gzip, Snappy, and Zstandard) that reduces downloading time and cloud egress fees. Additionally, Streaming also supports various hashing algorithms(SHA2, SHA3, MD5, xxHash, etc.) that ensures data integrity through cryptographic and non-cryptographic hashing algorithm.
+
+Let's jump right into an example on how to convert a raw dataset into a streaming format and load the same streaming format dataset for model training.
 
 ## Writing a dataset to streaming format
 
-This guide shows you how to use your custom Dataset with {class}`streaming.MDSWriter`, but the steps would also remain the same for another writer.
+This guide shows you how to use your custom Dataset with {class}`streaming.MDSWriter`, but the steps would remain the same for another writers.
```
Comment (knighton) [inline docs/source/getting_started/user_guide.md:22]: other writers

File: docs/source/getting_started/user_guide.md [hunk 6]
```diff
@@ -1,14 +1,25 @@
...
 
-For loading the dataset during model training, one needs to instantiate the {class}`streaming.Dataset` class with a dataset file path and provide that object to `dataset` parameter in PyTorch {class}`torch.utils.data.DataLoader` class. For the more information about dataset and its parameters, look at the API reference doc.
+For more information about writers and its parameters, look at the [API reference doc](../api_reference/).
+
+After the dataset has been converted to one of our streaming formats, one just needs to instantiate the {class}`streaming.Dataset` class by providing the dataset path of the streaming formats and use that dataset object in PyTorch {class}`torch.utils.data.DataLoader` class. For more information about `streaming.Dataset` and its parameters, look at the {class}`streaming.Dataset` API reference doc.
+
+Streaming supports various dataset compression formats(Brotli, Bzip2, Gzip, Snappy, and Zstandard) that reduces downloading time and cloud egress fees. Additionally, Streaming also supports various hashing algorithms(SHA2, SHA3, MD5, xxHash, etc.) that ensures data integrity through cryptographic and non-cryptographic hashing algorithm.
```
Comment (knighton) [inline docs/source/getting_started/user_guide.md:16]: space before (

File: docs/source/getting_started/user_guide.md [hunk 7]
```diff
@@ -40,27 +51,27 @@ class RandomClassificationDataset:
 
 There are a few parameters that need to be initialized before {class}`streaming.MDSWriter` gets called. Some of the parameters are optional, and others are required parameters. Let's look at each of them where we start with two required parameters.
 
-1. Provide the Local filesystem directory path to store the compressed dataset
+1. Provide the Local filesystem directory path to store the compressed dataset files.
```
Comment (knighton) [inline docs/source/getting_started/user_guide.md:54]: local filesystem

File: docs/source/getting_started/user_guide.md [hunk 8]
```diff
@@ -40,27 +51,27 @@ class RandomClassificationDataset:
...
+2. Provide the column field as `Dict[str, str]`, which maps a feature name or label name with a streaming supported encoding type.
     <!--pytest-codeblocks:cont-->
     ```python
     fields = {'x': 'pkl', 'y': 'pkl'}
     ```
 
 The below parameters are optional to {class}`streaming.MDSWriter`. Let's look at each one of them
 
-1. Provide a name of a compression algorithm; the default is `None`. We support families of compression algorithms, and all of them can be found here.
+1. Provide a name of a compression algorithm; the default is `None`. Streaming support families of compression algorithms such as `br`, `gzip`, `snappy`, `zstd`, and `bz2` with the level of compression.
```
Comment (knighton) [inline docs/source/getting_started/user_guide.md:68]: Streaming supports

**Review by knighton — APPROVED** [review]
Some minor quibbles about writing, looks great

File: docs/source/getting_started/quick_start.md (hunk 1 above)
Comment (karan6181) [reply docs/source/getting_started/quick_start.md]: Done in https://github.com/mosaicml/streaming/pull/33/commits/a634dbdbd8039c0755687d3cc8172cba15c977ed.

File: docs/source/getting_started/user_guide.md (hunk 2 above)
Comment (karan6181) [reply docs/source/getting_started/user_guide.md]: Done in https://github.com/mosaicml/streaming/pull/33/commits/a634dbdbd8039c0755687d3cc8172cba15c977ed.

File: docs/source/getting_started/user_guide.md (hunk 3 above)
Comment (karan6181) [reply docs/source/getting_started/user_guide.md]: Done in https://github.com/mosaicml/streaming/pull/33/commits/a634dbdbd8039c0755687d3cc8172cba15c977ed.

File: docs/source/getting_started/user_guide.md (hunk 4 above)
Comment (karan6181) [reply docs/source/getting_started/user_guide.md]: Done in https://github.com/mosaicml/streaming/pull/33/commits/a634dbdbd8039c0755687d3cc8172cba15c977ed.

File: docs/source/getting_started/user_guide.md (hunk 5 above)
Comment (karan6181) [reply docs/source/getting_started/user_guide.md]: Done in https://github.com/mosaicml/streaming/pull/33/commits/a634dbdbd8039c0755687d3cc8172cba15c977ed.

File: docs/source/getting_started/user_guide.md (hunk 6 above)
Comment (karan6181) [reply docs/source/getting_started/user_guide.md]: Done in https://github.com/mosaicml/streaming/pull/33/commits/a634dbdbd8039c0755687d3cc8172cba15c977ed.

File: docs/source/getting_started/user_guide.md (hunk 7 above)
Comment (karan6181) [reply docs/source/getting_started/user_guide.md]: Done in https://github.com/mosaicml/streaming/pull/33/commits/a634dbdbd8039c0755687d3cc8172cba15c977ed.

File: docs/source/getting_started/user_guide.md (hunk 8 above)
Comment (karan6181) [reply docs/source/getting_started/user_guide.md]: Done in https://github.com/mosaicml/streaming/pull/33/commits/a634dbdbd8039c0755687d3cc8172cba15c977ed.

**Pull request merged**